lib/pptx/pptx.go

```diff
@@ -0,0 +1,200 @@
+	builder.WriteString(`<cp:revision>1</cp:revision>`)
+	dateTime := time.Now().Format(time.RFC3339)
+	builder.WriteString(fmt.Sprintf(`<dcterms:created xsi:type="dcterms:W3CDTF">%s</dcterms:created>`, dateTime))
+	builder.WriteString(fmt.Sprintf(`<dcterms:modified xsi:type="dcterms:W3CDTF">%s</dcterms:modified>`, dateTime))
+	builder.WriteString(`<cp:category />`)
+	builder.WriteString(`</cp:coreProperties>`)
+
+	return builder.String()
+}
+
+func getAppXml(slides []*Slide, d2version string) string {
+	var builder strings.Builder
+	builder.WriteString(`<?xml version="1.0" encoding="UTF-8" standalone="yes"?>`)
+	builder.WriteString(`<Properties xmlns="http://schemas.openxmlformats.org/officeDocument/2006/extended-properties" xmlns:vt="http://schemas.openxmlformats.org/officeDocument/2006/docPropsVTypes">`)
+	builder.WriteString(`<TotalTime>1</TotalTime>`)
+	builder.WriteString(`<Words>0</Words>`)
+	builder.WriteString(`<Application>D2</Application>`)
+	builder.WriteString(`<PresentationFormat>On-screen Show (4:3)</PresentationFormat>`)
+	builder.WriteString(`<Paragraphs>0</Paragraphs>`)
+	builder.WriteString(fmt.Sprintf(`<Slides>%d</Slides>`, len(slides)))
+	builder.WriteString(`<Notes>0</Notes>`)
+	builder.WriteString(`<HiddenSlides>0</HiddenSlides>`)
+	builder.WriteString(`<MMClips>0</MMClips>`)
+	builder.WriteString(`<ScaleCrop>false</ScaleCrop>`)
+	builder.WriteString(`<HeadingPairs>`)
+	builder.WriteString(`<vt:vector size="6" baseType="variant">`)
+	builder.WriteString(`<vt:variant>`)
+	builder.WriteString(`<vt:lpstr>Fonts</vt:lpstr>`)
+	builder.WriteString(`</vt:variant>`)
+	builder.WriteString(`<vt:variant>`)
+	builder.WriteString(`<vt:i4>2</vt:i4>`)
+	builder.WriteString(`</vt:variant>`)
+	builder.WriteString(`<vt:variant>`)
+	builder.WriteString(`<vt:lpstr>Theme</vt:lpstr>`)
+	builder.WriteString(`</vt:variant>`)
+	builder.WriteString(`<vt:variant>`)
+	builder.WriteString(`<vt:i4>1</vt:i4>`)
+	builder.WriteString(`</vt:variant>`)
+	builder.WriteString(`<vt:variant>`)
+	builder.WriteString(`<vt:lpstr>Slide Titles</vt:lpstr>`)
+	builder.WriteString(`</vt:variant>`)
+	builder.WriteString(`<vt:variant>`)
+	builder.WriteString(fmt.Sprintf(`<vt:i4>%d</vt:i4>`, len(slides)))
+	builder.WriteString(`</vt:variant>`)
+	builder.WriteString(`</vt:vector>`)
+	builder.WriteString(`</HeadingPairs>`)
+	builder.WriteString(`<TitlesOfParts>`)
+	// number of entries, len(slides) + Office Theme + 2 Fonts
+	builder.WriteString(fmt.Sprintf(`<vt:vector size="%d" baseType="lpstr">`, len(slides)+3))
+	builder.WriteString(`<vt:lpstr>Arial</vt:lpstr>`)
+	builder.WriteString(`<vt:lpstr>Calibri</vt:lpstr>`)
+	builder.WriteString(`<vt:lpstr>Office Theme</vt:lpstr>`)
+	for _, slide := range slides {
+		builder.WriteString(fmt.Sprintf(`<vt:lpstr>%s</vt:lpstr>`, strings.Join(slide.BoardPath, "/")))
+	}
+	builder.WriteString(`</vt:vector>`)
+	builder.WriteString(`</TitlesOfParts>`)
+	builder.WriteString(`<Manager></Manager>`)
+	builder.WriteString(`<Company></Company>`)
```

Comment:
are these significant? we need these?

Reply:
They are part of the template, I can try to remove them

Reply:
sure they're part of the template but we ought to be examining the template used and not writing random tags to user ppts